lib/auth/auth.go

```diff
@@ -4329,6 +4330,18 @@ func (a *Server) CreateAccessRequestV2(ctx context.Context, req types.AccessRequ
 		return req, nil
 	}
 
+	// calculate the promotions
+	reqCopy := req.Copy()
+	promotions, err := modules.GetModules().GenerateAccessRequestPromotions(ctx, a.Services, reqCopy)
+	if err != nil {
+		// Do not fail the request if the promotions failed to generate.
+		// The request promotion will be blocked, but the request can still be approved.
+		log.WithError(err).Warn("Failed to generate access list promotions.")
+	}
+
+	// update the request with additional reviewers if possible.
+	updateAccessRequestWithAdditionalReviewers(ctx, req, a.AccessListClient(), promotions)
+
 	if err := a.verifyAccessRequestMonthlyLimit(ctx); err != nil {
```

Comment:
Should we calculate promotions after verifying the limit? No point of calculating it if the limit is exceeded.